treasury-fees-monitor/src/agent.js

```diff
@@ -0,0 +1,224 @@
+  return async function initialize() {
+    /* eslint-disable no-param-reassign */
+    // assign configurable fields
+    data.protocolName = config.protocolName;
+    data.protocolAbbreviation = config.protocolAbbreviation;
+    data.developerAbbreviation = config.developerAbbreviation;
+    data.findingLevelThresholds = config.thresholds;
+
+    const {
+      LendingPoolAddressesProvider: lendingPoolAddressesProvider,
+      LendingPool: lendingPool,
+      ProtocolDataProvider: protocolDataProvider,
+      PriceOracle: priceOracle,
+    } = config.contracts;
+    const provider = getEthersProvider();
+
+    // from the LendingPoolAddressesProvider, get the address of the LendingPool proxy contract
+    const lendingPoolAddressesProviderAbi = getAbi(lendingPoolAddressesProvider.abiFile);
+    data.lendingPoolAddressesProviderContract = new ethers.Contract(
+      lendingPoolAddressesProvider.address,
+      lendingPoolAddressesProviderAbi,
+      provider,
+    );
+    data.lendingPoolAddress = await data.lendingPoolAddressesProviderContract.getLendingPool();
+
+    // create a new interface for LendingPool
+    const lendingPoolAbi = getAbi(lendingPool.abiFile);
+    const lendingPoolInterface = new ethers.utils.Interface(lendingPoolAbi);
+
+    // retrieve the event signature for the FlashLoan event from the interface
+    const sigTypeFull = ethers.utils.FormatTypes.full;
+    data.flashLoanSignature = lendingPoolInterface.getEvent('FlashLoan').format(sigTypeFull);
+
+    const { address: protocolDataProviderAddress } = protocolDataProvider;
+    const protocolDataProviderAbi = getAbi(protocolDataProvider.abiFile);
+    const protocolDataProviderContract = new ethers.Contract(
+      protocolDataProviderAddress,
+      protocolDataProviderAbi,
+      provider,
+    );
+
+    // get all aToken addresses from the protocol data provider contract
+    data.tokenInfo = {};
+    const tokenAddresses = await protocolDataProviderContract.getAllReservesTokens();
+    // for each token address returned get its respective decimals
+    await Promise.all(tokenAddresses.map(async (token) => {
+      // token is an array where the first element is the token symbol and the second element is
+      // the address
+      const tokenAddress = token[1];
+      const tokenContract = new ethers.Contract(tokenAddress, DECIMALS_ABI, provider);
+      const tokenDecimals = await tokenContract.decimals();
+      data.tokenInfo[tokenAddress] = tokenDecimals.toString();
+    }));
+
+    // get the address of the price oracle
+    const priceOracleAddress = await data.lendingPoolAddressesProviderContract.getPriceOracle();
+    const priceOracleAbi = getAbi(priceOracle.abiFile);
+    data.priceOracleContract = new ethers.Contract(
+      priceOracleAddress,
+      priceOracleAbi,
+      provider,
+    );
+
+    // for each asset retrieve a spot price that will be used when computing statistics
+    const tokenAddressesList = tokenAddresses.map((token) => token[1]);
```

Comment:
A minor point, but you have iterated over these addresses once already.  You could wrap this Array creation into that block (lines 84-91) or just extract the token addresses that you have already collected with `Object.keys(data.tokenInfo)`.